app/components/DocumentListToolbar/DocumentListToolbar.jsx

```diff
@@ -47,89 +48,123 @@ class DocumentListToolbar extends React.Component {
+    this.goToPage = this.goToPage.bind(this)
+    this.goToNext = this.goToNext.bind(this)
+    this.handleInputChange = this.handleInputChange.bind(this)
+    this.handleInputKeyDown = this.handleInputKeyDown.bind(this)
+
     this.state = {
       goToPageValue: ''
     }
   }
 
-  goToPage(value) {
-    const {metadata, pageChangeHandler, route} = this.props
-    const parsedValue = Number.parseInt(value)
+  goToNext() {
+    this.goToPage(this.props.metadata.page + 1)
+  }
 
-    if (!metadata || typeof pageChangeHandler !== 'function') {
-      return null
+  goToPage(value) {
+    const {metadata, onPageChange} = this.props
+    const parsedValue = parseInt(value)
+
+    if (
+      metadata &&
+      typeof onPageChange === 'function' &&
+      parsedValue <= metadata.totalPages
+    ) {
+      this.setState({goToPageValue: ''})
+      onPageChange(parsedValue)
     }
+  }
 
-    // If the input is not a valid positive integer number, we return.
-    if (parsedValue.toString() !== value || parsedValue <= 0) return
-
-    // If the number inserted is outside the range of the pages available,
-    // we return.
-    if (parsedValue > metadata.totalPages) return
+  goToPrev() {
+    this.goToPage(this.props.metadata.page - 1)
+  }
 
-    const action = pageChangeHandler(parsedValue)
+  handleInputChange({target: {value}}) {
+    const parsedValue = parseInt(value)
 
-    // If the result of `pageChangeHandler` is a link, we redirect to it.
-    // If it is a function, we call it.
-    if (typeof action === 'string') {
-      route.history.push(action)
-    } else if (typeof action === 'function') {
-      action(parsedValue)
+    if (value === '' || (parsedValue.toString() === value && parsedValue > 0)) {
+      this.setState({goToPageValue: value})
     }
   }
 
-  handleGoToPage(event) {
-    const {goToPageValue} = this.state
-
-    event.preventDefault()
-
-    this.goToPage(goToPageValue)
+  handleInputKeyDown({key, target: {value}}) {
+    if (key === 'Enter') {
+      this.goToPage(value)
+    }
   }
 
   render() {
     const {
       children,
       metadata,
-      pageChangeHandler,
       selectedDocuments = [],
       showSelectedDocuments
     } = this.props
-    const {goToPageValue} = this.state
 
     if (!metadata) return null
 
-    const pagesObject = Array.apply(null, {
-      length: metadata.totalPages
-    }).reduce((result, _, index) => {
-      const page = index + 1
+    const {limit, offset, page, totalCount, totalPages} = metadata
+    const {goToPageValue} = this.state
 
-      result[page] = `Page ${page}`
+    const currentPageMin = (offset + 1).toLocaleString()
+    const currentPageMax = Math.min(offset + limit, totalCount).toLocaleString()
+    const numTotalDocuments = totalCount.toLocaleString()
+    const numSelectedDocuments = selectedDocuments.length.toLocaleString()
+
+    const pagesAroundCurrent = 2 // In each direction.
+    const ellipsisBefore = page > 2 + pagesAroundCurrent
```

Comment:
I'd call these be `showEllipsisBefore` and `showEllipsisAfter`.

Reply:
Done